src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/parenpad/InputParenPadCheckEmoji.java

```diff
@@ -0,0 +1,43 @@
+/*
+ParenPad
+option = (default)nospace
+tokens = (default)ANNOTATION, ANNOTATION_FIELD_DEF, CTOR_CALL, CTOR_DEF, DOT, \
+         ENUM_CONSTANT_DEF, EXPR, LITERAL_CATCH, LITERAL_DO, LITERAL_FOR, LITERAL_IF, \
+         LITERAL_NEW, LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_WHILE, METHOD_CALL, \
+         METHOD_DEF, QUESTION, RESOURCE_SPECIFICATION, SUPER_CTOR_CALL, LAMBDA, RECORD_DEF
+
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.whitespace.parenpad;
+
+import java.util.function.Consumer;
+
+public class InputParenPadCheckEmoji {
+    void emojiFamilyRunner1(Consumer<String> testRunner, String text) {
+        testRunner.accept("👩‍👩‍👧‍👧 " + text); // ok
+    }
+
+    void emojiFamilyWithSkinToneModifierRunner1(Consumer<String> testRunner, String text) {
+        testRunner.accept("👩🏻‍👩🏽‍👧🏾‍👦🏿 " + text); // ok
+    }
+
+    void emojiFamilyRunner2(Consumer<String> testRunner, String text) {
+        testRunner.accept("👩‍👩‍👧‍👧 " + text ); // violation
```

Comment:
It took me a while to understand where is problem

Can you use please put violation message in trailing comment?
https://github.com/checkstyle/checkstyle/blob/b9630a6475bef3e95430bfa7ac3cf5cc28f3a9e5/src/main/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/messages.properties#L15

Reply:
This:

```java
public class InputParenPadCheckEmoji {
    void emojiFamilyRunner1(Consumer<String> testRunner, String text) {
        testRunner.accept("👩‍👩‍👧‍👧 " + text); // ok
    }

    void emojiFamilyWithSkinToneModifierRunner1(Consumer<String> testRunner, String text) {
        testRunner.accept("👩🏻‍👩🏽‍👧🏾‍👦🏿 " + text); // ok
    }

    void emojiFamilyRunner2(Consumer<String> testRunner, String text) {
        testRunner.accept("👩‍👩‍👧‍👧 " + text ); // violation, ')' is preceded with whitespace
    }

    void emojiFamilyWithSkinToneModifierRunner2(Consumer<String> testRunner, String text) {
        testRunner.accept("👩🏻‍👩🏽‍👧🏾‍👦🏿 " + text ); // violation, ')' is preceded with whitespace
    }

    void emojiFamilyRunner3(Consumer<String> testRunner, String text) {
        testRunner.accept( "👩‍👩‍👧‍👧 " + text); // violation, '(' is followed by whitespace
    }

    void emojiFamilyWithSkinToneModifierRunner3(Consumer<String> testRunner, String text) {
        testRunner.accept( "👩🏻‍👩🏽‍👧🏾‍👦🏿 " + text); // violation, '(' is followed by whitespace
    }

    void emojiFamilyWithSkinToneModifierRunnerTricky(Consumer<String> testRunner, String text) {
        testRunner.accept( "👩🏻‍👩🏽‍👧🏾‍👦🏿 "  + "ab cdefg" + "👩🏽‍👧🏾‍👦🏿 " + text ); // 2 violations
    } // '(' is followed by whitespace, ')' is preceded with whitespace ^
}
```

Causes issues with inline config parser:

```bash
Violation lines for /home/nick/IdeaProjects/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/parenpad/InputParenPadCheckEmoji.java differ.
unexpected (4): 25, 29, 33, 37
---
expected      : [41, 41]
but was       : [25, 29, 33, 37, 41, 41]

	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyViolations(AbstractModuleTestSupport.java:427)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyWithInlineConfigParser(AbstractModuleTestSupport.java:270)


```

No change made to check. It seems like if trailing comment has anything extra, parser fails.

Comments have been placed under lines with violation.

Reply:
Can you try escape single quote?

Reply:
And remove `,`

Reply:
Please look at other places where message is used

Reply:
Mostly done, I could not make two different violations on the same line work, and could not find an example of this case.

Reply:
Two violation do not have this, we do not know how to even make it for now.
Ok to keep without message

Reply:
ok, done now. I think the comma with `'` afterwards was the problem.

Reply:
Awesome, thanks a lot